src/prepend_title_cell.py

```diff
@@ -0,0 +1,55 @@
+import json
+import nbformat as nbf
+import argparse
+import sys
+
+def add_ipynb_title(title: str, input_file: str, output_file: str, replace_first=False):
+    """Add or replaces the first cell of an ipython notebook with a markdown title
+
+    title (str): Markdown text to reside in the first cell
+    input_file (str): Path to the input file or template to be modified
+    output_file (str): Path to the write destination
+    replace_first (bool): If true, replaces the first cell of the input_file instead of prepending
+        a new cell
+
+    CLI use example:
+
+    python prepend_title.py -t '# New Header' -i 'Metrics Template.ipynb' -o 'my_notebook.ipynb'
+
+    """
+    # Ensure master template file won't be overwritten
+    if output_file.split('/')[-1]== 'Metrics Template.ipynb':
+        raise ValueError('Master Template overwrite is not permitted. Choose another output filepath')
+        sys.exit('1')
+    elif output_file.split('\\')[-1]== 'Metrics Template.ipynb': # ...Windows
+        raise ValueError('Master Template overwrite is not permitted. Choose another output filepath')
+        sys.exit('1')
+
+    # Create Title Cell
+    cell = nbf.v3.new_text_cell('markdown', title)
+
+    # Read in input_file ipynb
+    with open(input_file) as f:
+        data = json.load(f)
+    
+    met_nb = nbf.reads(json.dumps(data), as_version=4)
```

Comment:
What does this mean? (version 4)

Reply:
There are 4 versions of the ipynb json schema (see [nbformat](https://github.com/jupyter/nbformat/tree/master/nbformat)). v4 is the most recent structure (and the one most compatible with current ipynbs).  If you specify `as_version=3` the read JSON gets converted to a different dict structure.  

it's the difference between:

```json
# v4
{ 
  "cells": ["list_of_cells"]
}
```
and

```json
# v3
{ 
  "worksheets": [ 
    {"cells": ["list_of_cells"]}]
}
```

This just forces the dict format to be one I can handle downstream

Reply:
Makes sense